Create Meeting Notes 48 (#302)
cardano-foundation/CIPs: BiweeklyMeetings/2022-07-05.md

**Last Check**: Review of the PRIOR meetings Decisions  - if no objection, apply change (effectively a two week lag from decision to action, as a grace period)  

**New CIPs Review**: CIPs up for review should be looked over collectively, with discussion where needed. (on top of the asynchronous reviews)  

PR -> ‘Draft’: Needs format + approval.  
‘Draft’ -> ‘Proposed’: Needs a PLAN towards Active + implementation.  
‘Proposed’ -> ‘Active’:  Objective criteria as laid out observed, and consensus agreeing.   
**Current Discussions**: What the current CIPs discussions are on social media / forums / Discord.  

**Issues**: Review of the open Issues on the CIP GitHub 

**Close**: Recap of actions taken and decisions. List the CIPs that are due for review.  

## July 05 2022 Transcript
**Attending Editors**: Matthias Benkort, Sebastien Guillemot, ~Frederic Johnson~, ~Robert Phair~. 
**Guests**:

### Triage

#### PR274
[PR274 CIP-0059? | Terminology Surrounding Core Features](https://github.com/cardano-foundation/CIPs/pull/274)

> **Matthias** - :  Yeah, we can start already and go right in the first item, right, CIP 59, which was proposed by Jared, I think after a chat on a different CIP or different debates around Vasil versus Daedalus, and what do names mean exactly in Cardano. A lot of people seem to have gotten confused by the names being employed. So he wrote a small, informational CIP, there is not really much to debate in a way except maybe if people disagree with... I know the similar thing given to that and they want to change it, but since there is a type history already. It just explains where the name comes from, what they are and the rationale behind is why they were people like that. Like between era, and a phase, and a network and protocol. And he has also put up a nice table which summarizes all of that, which is nice, so it gives just a bit more clarity about naming.

Reviewer: ```suggestion
**Matthias** - :  Yeah, we can start already and go right in the first item, right, CIP 59, which was proposed by Jared, I think after a chat on a different CIP or different debates around Vasil versus Babbage, and what do names mean exactly in Cardano. A lot of people seem to have gotten confused by the names being employed. So he wrote a small, informational CIP, there is not really much to debate in a way except maybe if people disagree with it and want to change it, but since there is a track record already... It just explains where the name come from, what they are and the rationale behind is, why they were chosen like that. Like between era, and a phase, and a network and protocol. And he has also put up a nice table which summarizes all of that, which is nice, so it gives just a bit more clarity about naming.
```

> **Matthias** - :  And yeah, so I pointed to him last time a few things to do regarding the proposal of just the editorial stuff, putting a number, putting the readme in the correct folder. He did all of that and he already marked it as active as I suggested, just because as I said, it's an informational CIP explaining the terminology picked by the core team. It's from the core team itself and it's also recording or explaining things from the past six eras or so. So yeah, there is no point going to things like propose, and then active, and then whatever, unless people agree with that. Okay. 

Reviewer: ```suggestion
**Matthias** - :  And yeah, so I pointed to him last time a few things to do regarding the proposal of just the editorial stuff, putting a number, putting the readme in the correct folder. He did all of that and he already marked it as active as I suggested, just because as I said, it's an informational CIP explaining the terminology picked by the core team. It's from the core team itself and it's also recording or explaining things from the past six eras or so. So yeah, there is no point going to things like propose, and then active, and then whatever, unless someone disagree with that. Okay. 
```

**Sebastien** -, and your opinion maybe, if you are here.

**Sebastien** -:  I joined probably through the conversation, so I'm missing the beginning.

> **Matthias** - :  Yeah. Okay, sorry. We are talking about CIP 59, the terminologies surrounding co-features, which was proposed by Jared some time ago to clarify a bit of the terminology around the co-features. And I was saying, this one could probably just go as active right away because it's just explaining what's in there.

Reviewer: ```suggestion
**Matthias** - :  Yeah. Okay, sorry. We are talking about CIP 59, the terminologies surrounding core features, which was proposed by Jared some time ago to clarify a bit of the terminology around the core features. And I was saying, this one could probably just go as active right away because it's just explaining what's in there.
```

**Sebastien** -:  Yeah, I think so. I think that makes sense. I read through it and everything, and it made sense to me. So I don't mind both merging and marking it is actually the right away.

**Matthias** - :  We have a question. Oh sorry, that's some other stuff. So yeah okay, let's do that. I think we have reviewed it. I would just add my review now, that's all the... Yeah, All change has been made. Happy to approve. Happy to merge. Okay. Next one, on chain token metadata standard which has been reviewed for a while. We had a few remark questions. A lot of people were debating stuff in particular. The security of that, I think Michael Peyton Jones is usually against anything that is put on chain and it is based on minting transactions because indeed it has security issues or implications.

> **Matthias** - :  If we start heavily relying on people providing the metadata being the same as people minting the tokens, which has been generally the case in Cardano, but it's not a given. So I think he has a fair point. But yeah, the proposal has been eight all ever since, I think. It hasn't really moved, we are waiting for the author to come back to it. But yeah, I'm not sure if there is much to say on this one. Yeah. No remark, no question. I guess we can just leave it there.

Reviewer: ```suggestion
**Matthias** - :  If we start heavily relying on people providing the metadata being the same as people minting the tokens, which has been generally the case in Cardano, but it's not a given. So I think he has a fair point. But yeah, the proposal has been stalled ever since, I think. It hasn't really moved, we are waiting for the author to come back to it. But yeah, I'm not sure if there is much to say on this one. Yeah. No remark, no question. I guess we can just leave it there.
```

**Sebastien** -:  Yeah. I have something on this topic, but I'm going to have to address it in maybe a minute, if you just give me a minute.

> **Matthias** - :  Yeah, okay, okay. So depending on what you see, we'll see what to do with that one. Maybe it can also be picked up by someone else for a different standard or maybe reformulate the ID in a slightly different way. Yeah. And I don't know if there is any prior work also on that, I bet there is. So people putting all kind of metadata on chain. Now that has been the case for NFT, and I don't know if it has been the case for tokens themself or fungible tokens, at least I don't recall any project doing that.

Reviewer: ```suggestion
**Matthias** - :  Yeah, okay, okay. So depending on what you see, we'll see what to do with that one. Maybe it can also be picked up by someone else for a different standard or maybe reformulate the ID in a slightly different way. Yeah. And I don't know if there is any prior work also on that, I bet there is. So people putting all kind of metadata on chain. Now that has been the case for NFT, and I don't know if it has been the case for tokens themself -- I mean fungible tokens, at least I don't recall any project doing that.
```

**Sebastien** -:  Yeah. But I think it does tackle the issue because it makes other person writing the contract the same person that owns the token metadata, because it's no longer going to be in the transaction metadata field. So it couples the two together, is what you wanted to solve the issue, right?

**Matthias** - :  So you mean that, ah, it's captured in the policy itself.

**Sebastien** -:  Yeah. Exactly, yeah.

**Matthias** - :  Yeah? Okay, okay. I guess then you could make the argument then.

**Sebastien** -:  I think this would solve this issue. So I feel like we could revisit this if we just come up with the solution for the logging problem, which requires us to come up with a sip for how to come handle topics for log statements, which doesn't have right now. And then another sip for how we want Cardano to change to make this efficient in some way, shape or form.

> **Matthias** - :  Yeah, yeah, yeah. Because the logging, we have it at a stage, but you only have it, I think if the script is failing. So the script is happily successful, I'm not sure you can get any trace really out of it. But yeah, I guess the first good step is a CIP, which might requires another CIP, which would be a new product built in for doing this topic centric logging.

Reviewer: ```suggestion
**Matthias** - :  Yeah, yeah, yeah. Because the logging, we have it at a stage, but you only have it, I think, if the script is failing. So when the script is happily successful, I'm not sure you can get any trace really out of it. But yeah, I guess the first good step is a CIP, which might requires another CIP, which would be a new Plutus built-in for doing this topic centric logging.
```

**Sebastien** -:  Yeah. Which I think is the right solution because we could try matching a topic into the existing logging scheme, but it would be ugly. So probably it's better just come up with a new built in, come up with a solution to make it efficiently, and then we need some hassles [inaudible 00:10:52] developer to volunteer to do it, which is obviously the hardest part.

> **Matthias** - :  Yeah. I think you're going to get a lot of pushback on the logging idea because as I was saying, Plutus is just one big expression right now, there is no statement or really positive execution. So, when do you put a statement and where do you evaluate a statement is up to the interpreter at this stage, even though everything is tricked and it's pretty much, well, intuitive. That's probably getting the way of Bitwise optimizations and stuff like that. If you start requiring stuff to be done in sequence, and to be able to rely on that as a contract writer, that might be tricky. Perhaps actually, Les also has some inputs on that because from working with Plutus and all that stuff, that's maybe something you also thought of. I'm not sure if you want to.

Reviewer: ```suggestion
**Matthias** - :  Yeah. I think you're going to get a lot of pushback on the logging idea because as I was saying, Plutus is just one big expression right now, there is no statement or imperative execution. So, when do you put a statement and where do you evaluate a statement is up to the interpreter at this stage, even though everything is strict and it's pretty much, well, intuitive. That's probably getting in the way of low-level optimizations and stuff like that. If you start requiring stuff to be done in sequence, and to be able to rely on that as a contract writer, that might be tricky. Perhaps actually, Las also has some inputs on that because from working with Pluto / Plutarch, that's maybe something you also thought of. I'm not sure if you want to.
```

**Sebastien** -:  Yeah, it don't necessarily mean to hijack the CIP meeting to discuss this, but definitely Les, if you have some thoughts on this, or if you wanted to take this offline after this call and discuss this, I'd be really interested because this is something that we want for a few different projects on our side, and I'm sure this would unlock some other stack or other use cases like this token metadata on chain registry.

> **Matthias** - :  So sorry, I'm just writing on the chat. So yeah, let's just make a CIP offline discussions out of it for now and see. But yeah, I think my sentiment is that you would probably get some pushback from the Plutus core team because you're going to [inaudible 00:12:47].

Reviewer: ```suggestion
**Matthias** - :  So sorry, I'm just writing on the chat. So yeah, let's just make a CIP offline discussions out of it for now and see. But yeah, I think my sentiment is that you would probably get some pushback from the Plutus core team because you're going to introduce side-effects.

**Sebastien** -:  Yeah. Yeah. We get pushback on every feature, so it's just part of it.

#### PR137
[PR137 CIP-0038? | On-Chain Token Metadata Standard](https://github.com/cardano-foundation/CIPs/pull/137)

> **Matthias** - :  Yeah. It's part of the game, I guess. If you can argue it well enough, maybe, but yeah. Okay. So for this particular one, which was CIP 38, I think we can just pack it for now on the side, meaning that we're still waiting for the author. And if the author is not willing to move it forward, we won't be doing it for him basically. So yeah, there has been a few things said already, it's all recorded on the Pull Request as comments. So yeah, next steps are up to the author, not us.

Reviewer: ```suggestion
**Matthias** - :  Yeah. It's part of the game, I guess. If you can argue it well enough, maybe, but yeah. Okay. So for this particular one, which was CIP 38, I think we can just park it for now on the side, meaning that we're still waiting for the author. And if the author is not willing to move it forward, we won't be doing it for him basically. So yeah, there has been a few things said already, it's all recorded on the pull request as comments. So yeah, next steps are up to the author, not us.
```

#### PR151
[PR151 CIP-0030? | Events API](https://github.com/cardano-foundation/CIPs/pull/151)

**Matthias** - :  Which brings us to the next one, CIP 30, which is, I guess in a similar situation. I think we talked about that one several times in the past. Also, it was initially part of CIP 30, it was then extracted when CIP 30 was merged because there were still ongoing discussions and people were not sure whether or not they wanted it. I think there are a few wallets now that supports events as part of, not about CIP 30, but as part of this wallets extension work. And there is, I think, even a website put by this is part, right, with the state of what people support through the extension, right? Some things are labeled as experimental and I recall there was something related to events in there.

**Sebastien** -:  Yeah. So we put the old event that Nami had on the initial release, which is slightly different than what's in the events API PR, because the events API PR takes what Nami had originally and adds just on top of it.

> **Matthias** - :  So I'm just trying to find, okay, the website. Here is there just for linking it in the chats for people who wonder, so this is what I was mentioning.

Reviewer: ```suggestion
**Matthias** - :  So I'm just trying to find, okay, the website. Here is there just for linking it in the chats for people who wonder, so this is what I was mentioning ( https://www.cardano-caniuse.io/ ).
```

**Sebastien** -:  Yeah. And I still think we should have some event scheme. It just hasn't been a priority for a new one, which is why the CIP stalled. I do think it's important to have it.

**Matthias** - :  Yeah, okay. So I guess when the need will arise again, that CIP will just be either resurrected or superseded by some other CIP. But at this stage, the proposal is pretty dormant and waiting for the author. So, also not much to talk about, I guess.

#### PR163
[PR163 CIP-0037? | Dynamic Saturation based on Pledge](https://github.com/cardano-foundation/CIPs/pull/163)

**M. Àngels** -:  I added these few PRs to the agenda because we have been waiting for the author for several months. So I was wondering what to do in these kind of cases that the authors do not respond.

> **Matthias** - :  That's a good point. We could also close the PR after all, if there's some inactivity. People are still free to come back and ask to reopen them or comment on them saying, "Hey, I'm here. I have changes," or make a new PR, but I think it's not a big deal. But if we do that, I would also first, maybe mark it as a change in our CIP process, so make an update on, CIP 1, just say, "After X months of inactivity, we'll just close the Pull Request for keeping things tidy." That doesn't mean this proposal is rejected or anything, it just means it's inactive and we need to put some order on the ripple.

Reviewer: ```suggestion
**Matthias** - :  That's a good point. We could also close the PR after all, if there's some inactivity. People are still free to come back and ask to reopen them or comment on them saying, "Hey, I'm here. I have changes," or make a new PR, but I think it's not a big deal. But if we do that, I would also first, maybe mark it as a change in our CIP process, so make an update on, CIP 1, just say, "After X months of inactivity, we'll just close the Pull Request for keeping things tidy." That doesn't mean this proposal is rejected or anything, it just means it's inactive and we need to put some order on the repo.
```

**M. Àngels** -:  Do you think it's worth it to make that change on CIP 1?

**Matthias** - :  Yeah. I mean just making it in CIP 1 to be upfront saying, "This is what the process is," and at least people can also disagree with process as we change it, right, it's important, I think, to leave also the conversation open for that. But yeah, I would definitely say after, I don't know, four to six months of inactivity, which is quite a lot, we just close the Pull Request and people are free to come back with a new Pull Request if they still want to promote or propose their changes.

**M. Àngels** -:  I think it's good because in that way, we have a process and we don't need to revisit those proposals, and they are not waiting for the author forever.

> **Matthias** - :  Yeah. Yeah. And we avoid having too many Pull Requests stacking up, which are just waiting for author, waiting for author.

Reviewer: ```suggestion
**Matthias** - :  Yeah. Yeah. And we avoid having too many Pull Requests stacking up, which are just waiting for author.
```

### Review 

#### PR226
[PR226 CIP-0044? | Additional factors for NFT market verification](https://github.com/cardano-foundation/CIPs/pull/226)

**Matthias** - :Okay. So CIP 44 is next, and is also one we are waiting for NFT marketplaces feedback. Well, at least it is the main one.

**Sebastien** -:  Yeah. I think the message asking for feedback was sent out a day or two, right? So I think we're still waiting.

**Matthias** - :  So there's probably not much to do today on this one. Yeah. As we said last week, the ID looks good overall, but we had the question on whether or not this is useful for NFT marketplaces or endorsed by NFT marketplaces because it has to in order to become active. So we better include them early on in the feedback loop before merging or yeah, going with the proposed CIP. 

#### PR242
[PR242 CIP-0050? | Shelleys Voltaire decentralization update weekly update](https://github.com/cardano-foundation/CIPs/pull/242)

> **Matthias** - :So I guess we can just move to the next and last one on the agenda. Our favorite, CIP 50. I haven't got time to go through all the comments since last week, there has been a lot discussed. Again, there is a new member, I would say, in the discussions now, Robert, who is a mathematician working at Tesla, from what I got, who has got an interest in the CIP. So there has been quite a bunch of conversations between yeah, Rob and Michael, fortunately on the GitHub this time, on the GitHub issue. We also have created a physical channel for that one as we've been discussing last week, so a few more discussions have been going on here as well.

Reviewer: ```suggestion
**Matthias** - :So I guess we can just move to the next and last one on the agenda. Our favorite, CIP 50. I haven't got time to go through all the comments since last week, there has been a lot discussed. Again, there is a new member, I would say, in the discussions now, Robert, who is a mathematician working at Tesla, from what I got, who has got an interest in the CIP. So there has been quite a bunch of conversations between yeah, Rob and Michael, fortunately on GitHub this time, on the GitHub issue. We also have created a Discord channel for that one as we've been discussing last week, so a few more discussions have been going on here as well.
```

**Matthias** - :  Yeah. So beyond that, I haven't gone through all the comments. We could go through the comments as part of this current biweekly as well, since we have a bit of time left, or we could leave that for another time and try to do again a summary of the conversation that had been happening. Any thoughts on that?

**Sebastien** -:  Yeah. For this one, I feel like the summaries are nice, but I feel this one needs more than just a summary.

> **Matthias** - :  Of course. And as I said last time, I think this one can only move forward. At some point with regards the participation of IOG research. Without that, I'd say it's very difficult for that proposal to make concrete steps. And yeah, we have, I think two, yeah, Christopher and Colette. I think you are here both of you probably for the CIP 50, so maybe you have all sorts of stuff you want to discuss or say, please bring them up. Yeah, so I guess same situation for everyone. So I don't really have anything to add mainly in this call. We're talking about safety process in general, okay. And okay, so both of you. So is there any other items then we could discuss today before we wrap it up on maybe CIP 58? I don't think we have talked about that one yet. It was proposed and then proposed again in a new format, I think. So it's not from the same author, but it's from the same team as far as I understood, so it's superseding the previous proposal. So the first proposal was... Where it is? I'd lost it now.

Reviewer: It's always weird in the transcript when I am responding to the chat, but the chat isn't visible in the transcript. So it just looks like I am babbling stuff at random :shrug:

> **Matthias** - :  Okay, it was PR number 268 and the second proposal was PR 283, which is a new Plutus set of built-ins about doing Bitwise operations in Plutus. So manipulations on bites, which is quite common to do on low level stuff. It's been missing from Plutus so far, there is very pro-support from manipulating bite string in general and doing Bitwise operations in Plutus. Although it can be quite interesting because of the, well, performance aspect it can also bring in the picture. So it's a bit risky these things usually. So I haven't gone through the proposal yet, but I hope that it's a balance reducing a somewhat safe API for doing Bitwise operation and not allow people to shoot themselves in the foot with this stuff. I see it's proposed by the mLab's team and it has also been discussed with the Plutus team quite extensively, it seems. So this doesn't come out of the blue basically. So I think it'll be good to have it for review next time so that we get some good time to go through it, have a few comments and re-bring the spotlight to it on the next biweekly.

Reviewer: ```suggestion
**Matthias** - :  Okay, it was PR number 268 and the second proposal was PR 283, which is a new Plutus set of built-ins about doing bitwise operations in Plutus. So manipulations on bits, which is quite common to do on low level stuff. It's been missing from Plutus so far, there is very no support from manipulating byte strings in general and doing bitwise operations in Plutus. Although it can be quite interesting because of the, well, performance aspect it can also bring in the picture. So it's a bit risky these things usually. So I haven't gone through the proposal yet, but I hope that it's introducing a somewhat safe API for doing bitwise operations and not allow people to shoot themselves in the foot with this stuff. I see it's proposed by the MLab's team and it has also been discussed with the Plutus team quite extensively, it seems. So this doesn't come out of the blue basically. So I think it'll be good to have it for review next time so that we get some good time to go through it, have a few comments and re-bring the spotlight to it on the next biweekly.
```

**Matthias** - :  Because a rep needs to be paid by an owner?

**Sebastien** -:  Yeah. So it used to be free and then they changed it to paid, which is why it all links up probably.

**Matthias** - :  Okay. So at the same time, it's also filled to the current PR, so perhaps that's one good use of an issue. That's the other rep link, it doesn't work anymore, fine. But at least this PR fixes already a dead link, which is good in itself, which is comment to that actually. Okay. Nope, doesn't work. So I will open an issue following that. And where is actually the comment about rep link? Here. Rep link eight. So rep eight link doesn't work more. Okay. So this one is done. Do we have any small PR like that or should we just wrap it up for today? Or anything else people want to bring forward? Nope? Oh sorry, I've got a phone call. Okay, sorry about that. 

### Discussions
N/A

### Issues
N/A

### Close 

So let's wrap it up, maybe?

> **Matthias** - :  So we went through CIP 59 today, the terminology that runs core features. We agreed to merge it as active, as it is informal CIP bringing clarity on naming and coming from the core team directly. Then we've looked at CIP 38, CIP 30, CIP 37, I think that's the only three ones, which are waiting for an update from the author. We discussed that for those CIP that has been idle and dormant for a few months, it might be good actually to just close the Pull Requests and well, let the author come back with either a new Pull Request or an update if they do. But before doing that or in parallel of doing that, also specify that, "In the CIP 1, as a small change or in the process," so that we are upfront with what's going to happen. And it'll help us keep the repository tidy.

Reviewer: ```suggestion
**Matthias** - :  So we went through CIP 59 today, the terminology that runs core features. We agreed to merge it as active, as it is informal CIP bringing clarity on naming and coming from the core team directly. Then we've looked at CIP 38, CIP 30, CIP 37, I think that's the only three ones, which are waiting for an update from their authors. We discussed that for those CIPs that have been idle and dormant for a few months, it might be good actually to just close the Pull Requests and well, let the authors come back with either a new Pull Request or an update if they do. But before doing that or in parallel of doing that, also specify that, "In the CIP 1, as a small change or in the process," so that we are upfront with what's going to happen. And it'll help us keep the repository tidy.
```
